### ISSUE TITLE
enh: support `Expr.map_batches`

### ISSUE THREAD

**Issue by MarcoGorelli** [narwhals-dev/narwhals#1226]
use-case: https://github.com/narwhals-dev/narwhals/issues/1225

**Comment by vincentarelbundock** [comment]
I'm very excited to see this on the roadmap. Thanks for considering!

Would there be interest in adding `map_elements()` too? As you know, many functions must be applied element-wise, even if that is usually inefficient.

**Comment by vincentarelbundock** [comment]
Of course, there are easy workarounds for that, so I would totally understand if that's where you want to draw the line.

**Comment by MarcoGorelli** [comment]
thanks!

do you have a use-case for `map_elements` in `marginaleffects`?

**Comment by vincentarelbundock** [comment]
Formatting using custom f-strings is on the roadmap for Polars, but I don't think it is implemented yet: https://github.com/pola-rs/polars/issues/7133

I was using this instead:

```python
pl.col(col).map_elements(lambda x: f"{x:.3g}", return_dtype=pl.Utf8)
```

But as I said above, it's easy to find a workaround. So feel free to ignore if you think this is not generally useful, or bad use of the method.

**Comment by DeaMariaLeon** [comment]
I would like to work on this.

**Comment by MarcoGorelli** [comment]
yay thanks @DeaMariaLeon !

@vincentarelbundock if custom f-strings are on polars roadmap anyway, then perhaps we can wait a bit longer before introducing `map_elements` to Narwhals, I'm a little hesitant about that one

**Comment by DeaMariaLeon** [comment]
Do we want to include this parameter?
`is_elementwise`
"If set to true this can run in the streaming engine, but may yield incorrect results in group-by. Ensure you know what you are doing!" 
Taken from [here](https://github.com/pola-rs/polars/blob/main/py-polars/polars/expr/expr.py#L4315C9-L4317C79)

**Comment by MarcoGorelli** [comment]
i wouldn't think so, I think the only one we need is `return_dtype`

**Comment by FBruzzesi** [comment]
Closed by #1269
